### PR TITLE
pyoxisizer: init at 0.10.3

### LINKED ISSUE #104218: PyOxidizer
**Project description**
Rust tool for producing binaries which embed the Python runtime.

**Metadata**

* homepage URL: https://github.com/indygreg/PyOxidizer
* source URL: https://github.com/indygreg/PyOxidizer
* license: mpl-2.0
* platforms: all

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

Closes  #104218
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

CC @layday for testing

Metadata for this package is missing as I don't know what platforms it works on and also because I don't want to take maintainership. Maybe @layday would like to be maintainer for this?

### CODE REVIEW

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "indygreg";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "07hd00nyg1mbjlg8241ci5w6clqchcrlmld2vy4nca26j5b4k46k";
+  };
+
+  cargoSha256 = "14k81n56w29fdwm1bwbsmb3p7nawjbvfrqg2p74a0bz33hq5iby3";
+  depsBuildBuild = [ openssl pkgconfig python38 ];
+  doCheck = false;
```
Comment (Emantor) [inline pkgs/development/tools/pyoxidizer/default.nix:22]: The whole `meta` section is missing i.e.
```
  meta = with stdenv.lib; {
    description = "A utility for producing binaries that embed Python";
    homepage = "https://github.com/indygreg/PyOxidizer";
    license = licenses.mpl20;
    maintainers = with maintainers; [ ??? ];
    platforms = ???
```
You'll also need to add yourself as a maintainer to the `maintainers/maintainer-list.nix`, [manual](https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes) for further information on the meta attribute.

**Review by Emantor — CHANGES_REQUESTED** [review]
- [x] builds
```
1 package built:
pyoxidizer
```
- [x] binary works in so far that the help text is output
- [x] license is correct (in suggested meta tag)
- [ ] `bin` includes a binary named release, what does this do?

**Comment by layday** [comment]
Thanks for tagging me. I don’t think `pyoxidizer build` will work on Darwin without pulling in the SDK - the runtime depends on several of Apple’s frameworks.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by matthiasbeyer** [comment]
> . I don’t think `pyoxidizer build` will work on Darwin without pulling in the SDK - the runtime depends on several of Apple’s frameworks.

Feel free to send me patches for this, I don't have the infrastructure to test on Apple.

**Comment by matthiasbeyer** [comment]
@layday feel also free to send me a patch taking maintainership for this. I don't want to be maintainer, I just wanted to help to close #104218

**Comment by layday** [comment]
I would not be comfortable taking on maintenance of a package in a language I am unfamiliar with.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, pkgconfig
+, python38
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:3]: Do we need to pin python38 here?

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "07hd00nyg1mbjlg8241ci5w6clqchcrlmld2vy4nca26j5b4k46k";
+  };
+
+  cargoSha256 = "14k81n56w29fdwm1bwbsmb3p7nawjbvfrqg2p74a0bz33hq5iby3";
+  depsBuildBuild = [ openssl pkgconfig python38 ];
+  doCheck = false;
+
+  meta = with stdenv.lib; {
+    description = "A utility for producing binaries that embed Python";
+    homepage = "https://github.com/indygreg/PyOxidizer";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:26]: Maintainer is missing.

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "07hd00nyg1mbjlg8241ci5w6clqchcrlmld2vy4nca26j5b4k46k";
+  };
+
+  cargoSha256 = "14k81n56w29fdwm1bwbsmb3p7nawjbvfrqg2p74a0bz33hq5iby3";
+  depsBuildBuild = [ openssl pkgconfig python38 ];
+  doCheck = false;
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:24]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "indygreg";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "07hd00nyg1mbjlg8241ci5w6clqchcrlmld2vy4nca26j5b4k46k";
+  };
+
+  cargoSha256 = "14k81n56w29fdwm1bwbsmb3p7nawjbvfrqg2p74a0bz33hq5iby3";
+  depsBuildBuild = [ openssl pkgconfig python38 ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:21]:
```suggestion
  depsBuildBuild = [ openssl pkg-config python38 ];
```

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:2]:
```suggestion
, pkg-config
```

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:1]:
```suggestion
{ lib
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/pyoxidizer/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/pyoxidizer/default.nix:3]: Please rather use python3.

File: pkgs/development/tools/pyoxidizer/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "indygreg";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "07hd00nyg1mbjlg8241ci5w6clqchcrlmld2vy4nca26j5b4k46k";
+  };
+
+  cargoSha256 = "14k81n56w29fdwm1bwbsmb3p7nawjbvfrqg2p74a0bz33hq5iby3";
+  depsBuildBuild = [ openssl pkg-config python38 ];
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pyoxidizer/default.nix:22]: Please add a comment why you disabled tests.

**Pull request closed without merging**